flink-connector-mongodb-cdc/src/main/java/com/alibaba/ververica/cdc/connectors/mongodb/table/MongoDBConnectorDeserializationSchema.java

```diff
@@ -0,0 +1,721 @@
+                "Unable to convert to float from unexpected value '"
+                        + docObj
+                        + "' of type "
+                        + docObj.getBsonType());
+    }
+
+    private LocalDate convertInstantToLocalDate(Instant instant) {
+        return convertInstantToLocalDateTime(instant).toLocalDate();
+    }
+
+    private LocalTime convertInstantToLocalTime(Instant instant) {
+        return convertInstantToLocalDateTime(instant).toLocalTime();
+    }
+
+    private LocalDateTime convertInstantToLocalDateTime(Instant instant) {
+        return instant.atZone(SYSTEM_DEFAULT_ZONE).toLocalDateTime();
+    }
+
+    private Instant convertToInstant(BsonTimestamp bsonTimestamp) {
+        return Instant.ofEpochSecond(bsonTimestamp.getTime() * 1000);
+    }
+
+    private Instant convertToInstant(BsonDateTime bsonDateTime) {
+        return Instant.ofEpochMilli(bsonDateTime.getValue());
+    }
+
+    /**
+     * A conversion from DateType to int describes the number of days since epoch.
+     *
+     * @see org.apache.flink.table.types.logical.DateType
+     */
+    private int convertToDate(BsonValue docObj) {
+        if (docObj.isDateTime()) {
+            Instant instant = convertToInstant(docObj.asDateTime());
+            return (int) convertInstantToLocalDate(instant).toEpochDay();
+        }
+        if (docObj.isTimestamp()) {
+            Instant instant = convertToInstant(docObj.asTimestamp());
+            return (int) convertInstantToLocalDate(instant).toEpochDay();
+        }
+        throw new IllegalArgumentException(
+                "Unable to convert to date from unexpected value '"
+                        + docObj
+                        + "' of type "
+                        + docObj.getBsonType());
+    }
+
+    /**
+     * A conversion from TimeType to int describes the number of milliseconds of the day.
+     *
+     * @see org.apache.flink.table.types.logical.TimeType
+     */
+    private int convertToTime(BsonValue docObj) {
+        if (docObj.isDateTime()) {
+            Instant instant = convertToInstant(docObj.asDateTime());
+            return convertInstantToLocalTime(instant).toSecondOfDay() * 1000;
+        }
+        if (docObj.isTimestamp()) {
+            Instant instant = convertToInstant(docObj.asTimestamp());
+            return convertInstantToLocalTime(instant).toSecondOfDay() * 1000;
+        }
+        throw new IllegalArgumentException(
+                "Unable to convert to time from unexpected value '"
+                        + docObj
+                        + "' of type "
+                        + docObj.getBsonType());
+    }
+
+    private TimestampData convertToTimestamp(BsonValue docObj) {
+        if (docObj.isDateTime()) {
+            return TimestampData.fromEpochMillis(docObj.asDateTime().getValue());
+        }
+        if (docObj.isTimestamp()) {
+            return TimestampData.fromEpochMillis(docObj.asTimestamp().getTime() * 1000);
```

Comment:
1. 溢出问题多谢指正
2. Bson Timestamps只能精确到秒，increment只是为了保证唯一性的一个递增值，并不能作为具体时间间隔的参考。

[Bson Types](https://docs.mongodb.com/manual/reference/bson-types/)
```
Timestamps
BSON has a special timestamp type for internal MongoDB use and is not associated with the regular Date type. 
This internal timestamp type is a 64 bit value where:

the most significant 32 bits are a time_t value (seconds since the Unix epoch)
the least significant 32 bits are an incrementing ordinal for operations within a given second.

Within a single mongod instance, timestamp values are always unique.
```